drf_spectacular/plumbing.py

```diff
@@ -1126,6 +1126,23 @@ def _get_type_hint_origin(hint):
         return origin, args
 
 
+def _resolve_typeddict(hint):
+    """resolve required fields for TypedDicts if on 3.9 or above"""
+
+    required = None
+
+    if sys.version_info >= (3, 9):
+        required = [h for h in hint.__required_keys__]
+
+    return build_object_type(
+        properties={
+            k: resolve_type_hint(v) for k, v in get_type_hints(hint).items()
+        },
+        required=required,
+        description=inspect.cleandoc(hint.__doc__ or ""),
```

Comment:
I think it might be beneficial to use `plumbing.get_doc(hint)`. there is some additional cleanup going on.

Reply:
👍🏼  changed!